replications/internal/store.go

```diff
@@ -0,0 +1,256 @@
+package internal
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb/v2/kit/platform"
+	ierrors "github.com/influxdata/influxdb/v2/kit/platform/errors"
+	"github.com/influxdata/influxdb/v2/sqlite"
+	"github.com/mattn/go-sqlite3"
+)
+
+var errReplicationNotFound = &ierrors.Error{
+	Code: ierrors.ENotFound,
+	Msg:  "replication not found",
+}
+
+func errRemoteNotFound(id platform.ID, cause error) error {
+	return &ierrors.Error{
+		Code: ierrors.EInvalid,
+		Msg:  fmt.Sprintf("remote %q not found", id),
+		Err:  cause,
+	}
+}
+
+type Store struct {
+	sqlStore *sqlite.SqlStore
+}
+
+func NewStore(sqlStore *sqlite.SqlStore) *Store {
+	return &Store{
+		sqlStore: sqlStore,
+	}
+}
+
+func (s *Store) Lock() {
+	s.sqlStore.Mu.Lock()
+}
+
+func (s *Store) Unlock() {
+	s.sqlStore.Mu.Unlock()
+}
+
+// ListReplications returns a list of replications matching the provided filter.
+func (s *Store) ListReplications(ctx context.Context, filter influxdb.ReplicationListFilter) (*influxdb.Replications, error) {
+	q := sq.Select(
+		"id", "org_id", "name", "description", "remote_id", "local_bucket_id", "remote_bucket_id",
+		"max_queue_size_bytes", "latest_response_code", "latest_error_message", "drop_non_retryable_data").
+		From("replications")
+
+	if filter.OrgID.Valid() {
```

Comment:
Previously the orgID was always included in the `WHERE` clause. This meant that an empty filter (like the service "open" process uses) would return an empty list, and the replications system would be broken after re-starting the server.